jetty-server/src/main/java/org/eclipse/jetty/server/CachedContentFactory.java

```diff
@@ -237,7 +255,7 @@ private HttpContent load(String pathInContext, Resource resource, int maxBufferS
                 {
                     String compressedPathInContext = pathInContext + format._extension;
                     CachedHttpContent compressedContent = _cache.get(compressedPathInContext);
-                    if (compressedContent == null || compressedContent.isValid())
+                    if (compressedContent == null || !compressedContent.isValid())
```

Comment:
Why was this changed?

Reply:
We only want to get the resource from the factory again if it is null or not valid.
The code doesn't make much sense the way it was, I think it was a mistake and someone forgot the `!`.